nltk/tokenize/punkt.py

```diff
@@ -1062,25 +1062,32 @@ def _col_log_likelihood(count_a, count_b, count_ab, N):
 
         p = count_b / N
         p1 = count_ab / count_a
-        p2 = (count_b - count_ab) / (N - count_a)
 
-        summand1 = (count_ab * math.log(p) +
-                    (count_a - count_ab) * math.log(1.0 - p))
+        p2_denom = N - count_a
+        if p2_denom == 0:
+            p2 = 1
+        else:
+            p2 = (count_b - count_ab) / p2_denom
+
+        print (p, p1, p2, N, count_a, count_b, count_ab)
+
+        summand1 = (count_ab * math.log(p + (p == 0)) +
```

Comment:
I would replace ``math.log(p + (p == 0))`` with ``math.log(p or 1)``

Alternatively, the whole term could be replaced to
```
count_ab * math.log(p) if p else 0
```